charts/catalog/templates/controller-manager-deployment.yaml

```diff
@@ -0,0 +1,75 @@
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: {{ template "fullname" . }}-controller-manager
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "fullname" . }}-controller-manager
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}-controller-manager
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      containers:
+      - name: controller-manager
+        image: {{ .Values.controllerManager.image }}
+        imagePullPolicy: {{ .Values.controllerManager.imagePullPolicy }}
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+        args:
+        - --port
+        - "8080"
+        - --service-catalog-api-server-url
+        {{- if .Values.apiserver.insecure }}
+        - http://{{ template "fullname" . }}-apiserver
+        {{- else }}
+        - https://{{ template "fullname" . }}-apiserver
+        {{- end }}
+        - -v
+        - "{{ .Values.controllerManager.verbosity }}"
+        ports:
+        - containerPort: 8080
+        volumeMounts:
+        - name: service-catalog-cert
+          mountPath: /etc/service-catalog-ssl
+          readOnly: true
+        readinessProbe:
+          httpGet:
+            port: 8080
+            path: /healthz
+          failureThreshold: 1
+          initialDelaySeconds: 10
```

Comment:
I'm not sure... but I think that might actually be irrelevant because what I've seen so far is that (regardless of which storage mode is used) the controller-manager will exit non-zero the first time it tries to start because it will time out while waiting for the apiserver to be both running and passing its own readiness checks. I don't think this is anything to be concerned with, as this seems to be a common pattern in the k8s world-- allowing components to fail at startup until they can successfully connect to whatever other thing(s) they depend on that might not be ready initially.

So... for now, I'd leave this probe as is, because even if the controller manager takes a couple tries to start, it's not this probe that's killing it.